modules/openthread/platform/radio.c

```diff
@@ -385,6 +393,25 @@ void transmit_message(struct k_work *tx_job)
 	radio_api->set_channel(radio_dev, channel);
 	radio_api->set_txpower(radio_dev, get_transmit_power_for_channel(channel));
 
+#if defined(CONFIG_OPENTHREAD_TIME_SYNC)
+	if (sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeIeOffset != 0) {
+		*(sTransmitFrame.mPsdu + sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeIeOffset) =
+			sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeSyncSeq;
+		uint8_t *time_ie =
+			sTransmitFrame.mPsdu + sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeIeOffset;
+		uint64_t offset_plat_time =
+			otPlatTimeGet() + sTransmitFrame.mInfo.mTxInfo.mIeInfo->mNetworkTimeOffset;
+
+		*time_ie = sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeSyncSeq;
```

Comment:
Doesn't this duplicate the write at line 398 (I like this approach better though)?